### ISSUE TITLE
Build tasks dont respect "haxe.executable" setting

### ISSUE THREAD

**Issue by binkiesjoris** [vshaxe/vshaxe#226]
We are running 2 different versions of haxe side by side for different projects. Haxe3 is the default install but we are running Haxe4 for new projects. In order to facilitate easy switchability between projects I'm trying to configure the workspace settings and/or build tasks in such a way that each project automatically uses the correct version of Haxe.

My first attempt was to simply add a new environment variable for Haxe4 on each developer pc and change the build task to use the following: "command": "${env:HAXE4PATH}haxe.exe", however this does not work for the display server (which still targets the default haxe, presumable due to the HAXEPATH environment variable pointing to Haxe3), resulting in code completion etc not working as it cant compile.

According to the [documentation](https://github.com/vshaxe/vshaxe/wiki/Configuration#haxe-executable) we can target a different Haxe executable for the display server AND build tasks using the "haxe.executable" setting, which reduce the variable we'd have to switch to one given we dont use the command as specified above. This would still allow me to set it on a per project bases. (It would require all developers to install to the same directory as VSC does not support environment variables in settings but this is an acceptable compromise)

Using the "haxe.executable" setting resolved my issue with the display server, but the default Haxe3 executable is still being used when building and not the one set in the haxe.executable setting. From my understanding of the documentation it is not supposed to work like this, or have I misunderstood?

Additionally, if I change the build task command from the default "haxe" value the terminal keeps popping up when a compile error occurs even though the presentation.reveal setting is set to "silent". This seems to indicate the problemMatcher doesnt match so changing the command isnt really an option either.

**Comment by Gama11** [comment]
Do you have custom task definitions in your `tasks.json`? `haxe.executable` can only be respected for tasks that are _generated_ by vshaxe. This includes auto-discovered HXML file tasks (basically HXML files in the project root directory) and tasks from the `settings.json` config.

Here's a simple example with working configuration which may be helpful to you: https://github.com/Gama11/HaxeRepro

**Comment by binkiesjoris** [comment]
Our tasks.json is the following (without the command change):
`{
	"version": "2.0.0",
    "tasks": 
    [
		{
			"label": "build",
			"command": "haxe",
			"args": ["build.hxml", "-debug"],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"problemMatcher": "$haxe",
			"presentation": {
				"echo": true,
				"reveal": "silent",
				"focus": false,
				"panel": "shared"
			}
        }
    ]
}`

We might have changed some things here by hand but I can try to see what happens if I let vshaxe generate it again, I'm not seeing much difference from your example though. Our build.hxml is in the project root as well but doesnt contain anything special.

**Comment by Gama11** [comment]
Yes, the first task example is just a custom tasks that calls `haxe` manually via it's `command`. Vshaxe knows nothing about it (and can't, due to how VSCode's API works). The second one on the other hand is created by vshaxe (indicated by `"type": "hxml"`), so it has more or less full control over it.

Note that it's good practice to remove the inserted `problemMatcher` array when customizing a task. It will just use the default problem-matchers then (which are the same), and you get a more concise `tasks.json`. And if we ever change the list of problem matchers on the vshaxe side, those changes will be applied automatically to your task. I think VSCode mostly inserts these for discoverability / to let you know that you can customize the problem matcher list.
